### ISSUE TITLE
Simple Echo example

### ISSUE THREAD

**Issue by BikeBr0** [jkdhn/mcumgr-dart#1]
I am evaluating this pluigin for use in a Flutter App. Do you have a simple example of issuing an ECHO command? I realize this plugin does not mean to support all mcumgr commands but a little example would help to expose what more work is needed on my side. For example: here is the code required to issue the ECHO command using Nordic's mcumgr for Android:

````
      String uuid = call.argument("uuid");
      BluetoothDevice bluetoothDevice = manager.getAdapter().getRemoteDevice(uuid);
      String message = call.argument("message");
      McuMgrTransport transport = new McuMgrBleTransport(context, bluetoothDevice);
      DefaultManager defaultManager = new DefaultManager(transport);

      defaultManager.echo(message, new McuMgrCallback<McuMgrEchoResponse>() {
        @Override
        public void onResponse(@NotNull McuMgrEchoResponse response) {
          result.success(new String(response.r.getBytes(StandardCharsets.UTF_8)));
        }

        @Override
        public void onError(@NotNull McuMgrException error) {
          result.error("MCUMGR_ERROR", "Echo command failed", error.getMessage());
        }
      });

````

**Comment by jkdhn** [comment]
To create a client, you need another package which provides Bluetooth connectivity.
I've used flutter_blue in the past, but that looks like it's unmaintained, flutter_reactive_ble might be a good alternative.

You need to discover the service (`8d53dc1d-1db7-4cd3-868b-8a527460aa84`) and characteristic (`da2e7828-fbce-4e01-ae9e-261174997c48`). Then, you subscribe to notifications from the characteristic and pass that stream to the Client constructor - the client will read responses from there. To send requests, it also needs a callback which writes a byte array to the characteristic. There's an example for that in the README.md.

Also remember to increase the MTU of the connection, the default is way too low and probably won't work.

After that, using the echo command looks like this:
```dart
final response = await client.echo("Hello World", Duration(seconds: 5));
```
This will throw an exception if the command times out or the device returns an error.

Only few commands are implemented yet (for example, I just added echo in 0.0.2), so depending on what you want to do, you might have to use `client.execute` (and/or create an issue). But everything that's needed for firmware uploads is there.

There's no full example yet, I'll probably write one for firmware uploads using flutter_reactive_ble this weekend.

**Comment by BikeBr0** [comment]
Thanks for getting back to me @jkdhn! I forged ahead and got echo working with the 0.0.1 library using flutter_blue (aside: I hope this is still maintained as it was less funky to use than flutter_reactive_ble). Here is my connect and sendEcho code:

```
Future connect() async {
    _isConnected = await _scanResult.device.connect(autoConnect: false, timeout: const Duration(seconds: 5)).then((value) => true);
    
    // Increase the Bluetooth MTU (this is important!)
    _scanResult.device.requestMtu(252);
    _scanResult.device.mtu.listen((mtuVal) {
      log('mtu: $mtuVal');
    });

    _services = await _scanResult.device.discoverServices();

    // Find the service and its characteristic
    final service = _services.singleWhere((e) => e.uuid == serviceUuid);
    final BluetoothCharacteristic characteristic = service.characteristics.singleWhere((e) => e.uuid == characteristicUuid);

    // Subscribe to messages from the device
    await characteristic.setNotifyValue(true);

    _client = Client(
      // Used to receive messages from the device
      input: characteristic.value,
      // Used to send messages to the device
      output: (value) => characteristic.write(value, withoutResponse: true),
    );
  }
```

```
  Future<Message> sendEchoCmd() async {
    Message msg = Message(op: Operation.write, group: 0, id: 0, flags: 0, data: CborMap({CborString("d"): CborString("Bye")}));
    return _client!.execute(msg, const Duration(seconds: 5));
  }
```

It is worth noting that awaiting the response of `requestMtu(252);` does not work (code just hangs). I'm not sure if that is a bug in this library or the device I am connecting to.

Edit: regardless of awaiting the response of requestMtu the code sometimes just hangs. Again, I'm not sure where this intermittent bug is occurring.
